File: webservices/src/main/java/fieg/externos/caixaboletoservice/consulta/dto/TituloDTO.java
package fieg.externos.caixaboletoservice.consulta.dto;

import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
public class TituloDTO {
    private String numeroDocumento;
    private String dataVencimento;
    private String valor;
    private String tipoEspecie;
    private String flagAceite;
    private String dataEmissao;
    private JurosMoraDTO jurosMora;
    private String valorAbatimento;
    private PosVencimentoDTO posVencimento;
    private String codigoMoeda;
    private PagadorDTO pagador;
    private MultaDTO multa;
    private List<DescontoDTO> descontos;
    private String valorIof;
    private String identificacaoEmpresa;
    private PagamentoDTO pagamento;
    private String codigoBarras;
    private String linhaDigitavel;
    private String url;
}

File: web/src/models/DTOs/ConsultaBoletoCaixa/TituloDTO.ts
import { DescontoDTO } from "./DescontoDTO";
import { JurosMoraDTO } from "./JurosMoraDTO";
import { MultaDTO } from "./MultaDTO";
import { PagadorDTO } from "./PagadorDTO";
import { PagamentoDTO } from "./PagamentoDTO";
import { PosVencimentoDTO } from "./PosVencimentoDTO";

export interface TituloDTO {
  numeroDocumento: string;
  dataVencimento: string;
  valor: string;
  tipoEspecie: string;
  flagAceite: string;
  dataEmissao: string;
  jurosMora: JurosMoraDTO;
  valorAbatimento: string;
  posVencimento: PosVencimentoDTO;
  codigoMoeda: string;
  pagador: PagadorDTO;
  multa: MultaDTO;
  descontos: DescontoDTO[];
  valorIof: string;
  identificacaoEmpresa: string;
  pagamento: PagamentoDTO;
  codigoBarras: string;
  linhaDigitavel: string;
  url: string;
}